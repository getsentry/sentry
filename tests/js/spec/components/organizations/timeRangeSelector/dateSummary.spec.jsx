import React from 'react';

import {mount} from 'sentry-test/enzyme';

import DateSummary from 'app/components/organizations/timeRangeSelector/dateSummary';

const start = new Date('2017-10-14T02:38:00.000Z');
const end = new Date('2017-10-17T02:38:00.000Z'); // National Pasta Day

describe('DateSummary', function() {
  let wrapper;
  const routerContext = TestStubs.routerContext();

  const createWrapper = (props = {}) =>
    mount(<DateSummary utc start={start} end={end} {...props} />, routerContext);

  it('renders', async function() {
    wrapper = createWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it('does not show times when it is midnight for start date and 23:59:59 for end date', function() {
    // Date Summary formats using system time
    // tests run on EST/EDT
    wrapper = createWrapper({
      start: new Date('2017-10-14T00:00:00.000-0400'),
      end: new Date('2017-10-17T23:59:59.000-0400'),
    });

    expect(wrapper.find('Time')).toHaveLength(0);
  });
});
