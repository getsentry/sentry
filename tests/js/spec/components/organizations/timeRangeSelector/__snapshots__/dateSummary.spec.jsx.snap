// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSummary renders 1`] = `
<DateSummary
  end={2017-10-17T02:38:00.000Z}
  start={2017-10-14T02:38:00.000Z}
  utc={true}
>
  <DateGroupWrapper
    hasTime={true}
  >
    <div
      className="css-920ayr-DateGroupWrapper eyxmd2r0"
    >
      <DateGroup>
        <div
          className="css-1i8b5kk-DateGroup eyxmd2r1"
        >
          <Date
            hasTime={true}
          >
            <div
              className="css-20umtq-Date eyxmd2r2"
            >
              Oct 13, 2017
              <Time>
                <div
                  className="css-ybzw55-Time eyxmd2r3"
                >
                  22:38
                </div>
              </Time>
            </div>
          </Date>
        </div>
      </DateGroup>
      <DateRangeDivider>
        <span
          className="css-1pxvoc9-DateRangeDivider eyxmd2r4"
        >
          to
        </span>
      </DateRangeDivider>
      <DateGroup>
        <div
          className="css-1i8b5kk-DateGroup eyxmd2r1"
        >
          <Date
            hasTime={true}
          >
            <div
              className="css-20umtq-Date eyxmd2r2"
            >
              Oct 16, 2017
              <Time>
                <div
                  className="css-ybzw55-Time eyxmd2r3"
                >
                  22:38
                </div>
              </Time>
            </div>
          </Date>
        </div>
      </DateGroup>
    </div>
  </DateGroupWrapper>
</DateSummary>
`;
